ddtrace/internal/telemetry/writer.py

```diff
@@ -268,8 +272,8 @@ def enable(self):
         if self.status == ServiceStatus.RUNNING:
             return
 
-        self.start()
         self._enabled = True
+        self.start()
```

Comment:
Here's the main fix. Set TelemetryWriter._enabled=true BEFORE starting the writer thread.